README.md

```diff
@@ -1,1218 +1,533 @@
+
+# Plugin configuration
+<Plugin disk>
+  Disk "/dev/sda"
+  IgnoreSelected false
+</Plugin>
+```
+
+### Important Settings
+
+- `Hostname`: Sets the hostname for collected data
+- `Interval`: Default interval between data collections (in seconds)
+- `ReadThreads`: Number of threads used for data collection
+- `WriteThreads`: Number of threads used for data dispatch
+
+## Usage
+
+### Starting collectd
+
+```bash
+# Start as daemon
+collectd
+
+# Run in foreground
+collectd -f
+
+# Test configuration
+collectd -t
+
+# Use alternate config file
+collectd -C /path/to/config
+```
+
+### Signals
+
+- `SIGINT`, `SIGTERM`: Graceful shutdown
+- `SIGHUP`: Restart daemon
+- `SIGUSR1`: Flush all plugins
+
+## Plugin System
+
+```mermaid
+graph TD
+    A[Input Plugins] -->|Collect| B[Core]
+    B -->|Process| C[Filter Plugins]
+    C -->|Transform| D[Output Plugins]
+    
+    subgraph Input Plugins
+    E[CPU]
+    F[Memory]
+    G[Disk]
+    H[Network]
+    end
+    
+    subgraph Output Plugins
+    I[RRDTool]
+    J[CSV]
+    K[Network]
+    L[Databases]
+    end
+```
+
+### Input Plugins
+
+- **System Metrics**
+  - cpu: CPU usage statistics
+  - memory: Memory usage statistics
+  - disk: Disk usage and I/O statistics
+  - interface: Network interface statistics
+  - load: System load statistics
+
+- **Application Metrics**
+  - apache: Apache web server statistics
+  - mysql: MySQL database statistics
+  - nginx: Nginx web server statistics
+  - postgresql: PostgreSQL database statistics
+  - java: JVM statistics via JMX
+
+### Output Plugins
+
+- **File-based Storage**
+  - csv: Write to CSV files
+  - rrdtool: Write to RRD files
+
+- **Network and Databases**
+  - network: Send data over the network
+  - write_graphite: Send to Graphite
+  - write_prometheus: Expose for Prometheus
+  - write_influxdb: Send to InfluxDB
+
+## Data Storage
+
+### RRD Storage
+
+```apache
+LoadPlugin rrdtool
+<Plugin rrdtool>
+  DataDir "/var/lib/collectd/rrd"
+  CacheFlush 120
+  WritesPerSecond 50
+</Plugin>
+```
+
+### CSV Storage
+
+```apache
+LoadPlugin csv
+<Plugin csv>
+  DataDir "/var/lib/collectd/csv"
+  StoreRates false
+</Plugin>
+```
+
+## Networking
+
+```mermaid
+graph TD
+    A[Collector] -->|Send| B[Network Plugin]
+    B -->|Encrypt| C[Network]
+    C -->|Decrypt| D[Receiver]
+    D -->|Process| E[Storage]
+    
+    subgraph Security
+    F[Authentication]
+    G[Encryption]
+    end
+    
+    F -->|Verify| B
+    G -->|Protect| B
+```
+
+### Network Configuration
+
+```apache
+LoadPlugin network
+<Plugin network>
+  <Server "192.168.1.101" "25826">
+    SecurityLevel Encrypt
+    Username "user"
+    Password "secret"
+  </Server>
+</Plugin>
+```
+
+## Security
+
+### Security Features
+
+- Authentication with username/password
+- Network traffic encryption
+- Access control lists
+- Plugin-specific security options
+
+## Integration
+
+### Supported Integrations
+
+- **Visualization**
+  - Grafana
+  - Kibana
+  - Custom dashboards
+
```

Comment:
I do not see any of this in the collectd repo...?

Grafana does include several collectd dashboard examples for collectd though => maybe this should have just links to things like this: https://grafana.com/search/?term=collectd&type=dashboard

Reply:
good point, none of them are part of collectd.